src/main/commandServer/settingsValidator.ts

```diff
@@ -0,0 +1,199 @@
+import { Settings } from '@/config/settings';
+
+type validationFunc = (desiredValue: string|boolean|Record<string, any>, errors: string[], fqname: string) => boolean;
```

Comment:
Odd that we're not passing in the old value (since we're walking the struct anyway); but I guess that's reasonable as we only allow very few whitelisted entries, each with a custom validation routine?

Reply:
The global input value and the reduce code was a holdover from the prototype. I'm now passing the current config in as well and having the current config, proposed config, and validator all walked in lockstep.

Also, I've lost the comment, but the reason why I set the config each time is that its declared with `let` and not `const` in `background.ts`, so it was possible for the `SettingsValidator` to be pointing at an obsolete config instance. With the new changes that's no longer an issue.